core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -79,6 +79,9 @@ private[ui] class StagePage(parent: StagesTab, store: AppStatusStore) extends We
     localityNamesAndCounts.sorted.mkString("; ")
   }
 
+  private def jobURL(request: HttpServletRequest, jobId: Int): String =
```

Comment:
It all looks fine except I don't see a lot of value in moving this into a method that's used once. It just moves the logic to a separate part of the file.

Reply:
Make sense. I have updated the code.